app/controllers/reactions_controller.rb

```diff
@@ -102,10 +102,11 @@ def create
     render json: { result: result, category: category }
   end
 
-  def cached_user_public_reactions(user)
-    Rails.cache.fetch("cached-user-reactions-#{user.id}-#{user.public_reactions_count}", expires_in: 24.hours) do
-      user.reactions.public_category
+  def cached_user_public_reactions(user, comment_ids)
+    cache = Rails.cache.fetch("cached-user-reactions-#{user.id}-#{user.public_reactions_count}", expires_in: 24.hours) do
+      JSON.parse(user.reactions.public_category.to_json)
     end
+    cache.select { |a| comment_ids.include?(a["reactable_id"]) }
```

Comment:
Your above comment makes sense. I think there is actually a bug here now that I think about it. We select the reactions where reactable_id is included in the comment_ids but we dont ensure that the reactions are for comments. I think we want to filter by reactable_type here as well otherwise we might get an article reaction in this group.

Reply:
Great point. I've added an additional `where` clause and updated this method for clarity.